e2e/suite_test.go

```diff
@@ -40,54 +33,74 @@ func Test_All(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	scenarios := scenario.AllScenarios()
 	if err := createMissingClusters(ctx, r, scenarios, &clusterConfigs); err != nil {
 		t.Fatal(err)
 	}
 
 	for _, e2eScenario := range scenarios {
-		e2eScenario := e2eScenario
```

Comment:
I think we need the copy here since the loop counter is being used within a closure, otherwise you'll get closures that always act on the last element in the slice

I can see this behavior showing up in the most recent run off this branch: 

```
=== CONT  Test_All/ubuntu1804
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/marinerv2-custom-sysctls
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/ubuntu2204-gpu-a10
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/ubuntu2204-gpu-a100
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-chrony-restarts
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/ubuntu2204-gpu-ncv3
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-arm64-airgap
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-airgap
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-gpu-azurecni
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-azurecni
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-wasm
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-custom-sysctls
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-gpu
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/marinerv2-gpu
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/ubuntu2204-artifact-streaming
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/azurelinuxv2-arm64
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image
=== CONT  Test_All/ubuntu2204ContainerdVersion
    suite_test.go:60: skipping scenario "ubuntu2204-gpu-a10": could not find image

```

Reply:
ah, this is interesting: https://go.dev/doc/faq#closures_and_goroutines

looks like this was fixed in 1.22, I guess you might as well bump us up while you're at it :)

Reply:
ah, good catch. I run 1.22 locally and didn't realise ADO uses a different version.

Reply:
I wrongly assumed that passing it as a function argument would be enough to avoid the issue.
But instead, it should be copied before a goroutine is created. And it's not obvious when it happens in the tests.